src/ParseNode.js

```diff
@@ -23,7 +24,7 @@ export default class ParseNode {
     constructor(
         type: string,       // type of node, like e.g. "ordgroup"
         value: mixed,       // type-specific representation of the node
```

Comment:
I just noticed this was `mixed`.  Hopefully we'll be able to make this more precise in the future.

Reply:
I agree. A lot of refactoring needs to be done before we can do that.